src/DockerRunner.jl

```diff
@@ -97,8 +98,10 @@ function DockerRunner(workspace_root::String;
         push!(workspaces, "binarybuilder_ccache" => "/root/.ccache")
     end
 
-    # Choose the shards we're going to mount
-    shards = choose_shards(platform; extract_kwargs(kwargs, (:preferred_gcc_version,:preferred_llvm_version,:bootstrap_list,:compilers))...)
+    if isnothing(shards)
+        # Choose the shards we're going to mount
+        shards = choose_shards(platform; extract_kwargs(kwargs, (:preferred_gcc_version,:preferred_llvm_version,:bootstrap_list,:compilers))...)
```

Comment:
Can we move this into the `kwarg` initialization above?  I'd rather avoid the "default value of `nothing` means initialize it halfway through the method" pattern if we can, and I don't think there's a hidden reason why `choose_shards()` must be run after the stuff above it.

Reply:
We don't know `kwargs` when we initialise `shards`, do we?